src/algorithm/regressor.hxx

```diff
@@ -7,51 +7,258 @@
-                           DataTpl<Scalar,Options,JointCollectionTpl> & data,
-                           const Eigen::MatrixBase<ConfigVectorType> & q)
+    typedef ModelTpl<Scalar,Options,JointCollectionTpl> Model;
+    typedef DataTpl<Scalar,Options,JointCollectionTpl> Data;
+    typedef typename Model::JointIndex JointIndex;
+    typedef typename Data::SE3 SE3;
+
+    typedef typename Data::Matrix3x Matrix3x;
+    typedef typename SizeDepType<4>::ColsReturn<Matrix3x>::Type ColsBlock;
+    
+    forwardKinematics(model,data,q.derived());
+    
+    // Computes the total mass of the system
+    Scalar mass = Scalar(0);
+    for(JointIndex i = 1; i < (JointIndex)model.njoints; ++i)
+      mass += model.inertias[(JointIndex)i].mass();
+    
+    const Scalar mass_inv = Scalar(1)/mass;
+    for(JointIndex i = 1; i < (JointIndex)model.njoints; ++i)
+    {
+      const SE3 & oMi = data.oMi[i];
+      ColsBlock sr_cols = data.staticRegressor.template middleCols<4>((Eigen::DenseIndex)(i-1)*4);
+      sr_cols.col(0) = oMi.translation();
+      sr_cols.template rightCols<3>() = oMi.rotation();
+      sr_cols *= mass_inv;
+    }
+    
+    return data.staticRegressor;
+  }
+
+  namespace details {
+    // auxiliary function for bodyRegressor: bigL(omega)*I.toDynamicParameters().tail<6>() = I.inertia() * omega
+    template<typename Vector3Like>
+    inline Eigen::Matrix<typename Vector3Like::Scalar,3,6,PINOCCHIO_EIGEN_PLAIN_TYPE(Vector3Like)::Options>
+    bigL(const Eigen::MatrixBase<Vector3Like> & omega)
+    {
+      typedef typename Vector3Like::Scalar Scalar;
+      enum { Options = PINOCCHIO_EIGEN_PLAIN_TYPE(Vector3Like)::Options };
+      typedef Eigen::Matrix<Scalar,3,6,Options> ReturnType;
+
+      ReturnType L;
+      L <<  omega[0],  omega[1], Scalar(0),  omega[2], Scalar(0), Scalar(0),
+           Scalar(0),  omega[0],  omega[1], Scalar(0),  omega[2], Scalar(0),
+           Scalar(0), Scalar(0), Scalar(0),  omega[0],  omega[1],  omega[2];
+      return L;
+    }
+  }
+
+  template<typename MotionVelocity, typename MotionAcceleration, typename OutputType>
+  inline void
+  bodyRegressor(const MotionDense<MotionVelocity> & v, const MotionDense<MotionAcceleration> & a, const Eigen::MatrixBase<OutputType> & regressor)
+  {
+    PINOCCHIO_ASSERT_MATRIX_SPECIFIC_SIZE(OutputType, params, 6, 10);
+
+    typedef typename MotionVelocity::Scalar Scalar;
+    enum { Options = PINOCCHIO_EIGEN_PLAIN_TYPE(typename MotionVelocity::Vector3)::Options };
+
+    typedef Symmetric3Tpl<Scalar,Options> Symmetric3;
+    typedef typename Symmetric3::SkewSquare SkewSquare;
+    using ::pinocchio::details::bigL;
+
+    Eigen::MatrixBase<OutputType> & res = const_cast< Eigen::MatrixBase<OutputType> & >(regressor);
+
+    res.template block<3,1>(MotionVelocity::LINEAR,0) = a.linear() + v.angular().cross(v.linear());
+    const Eigen::Matrix<Scalar, 3, 1, Options> & acc = res.template block<3,1>(MotionVelocity::LINEAR,0);
```

Comment:
You are making a copy here for nothing.

Reply:
What? No. I am using a reference. This is a change I introduce with last commits following one of your previous remarks. `acc` is used later

Reply:
Yes, but a block is not an Eigen::Matrix, so it will allocate the memory. You have to use a block expression to address it.

Reply:
Mmh, I thought Eigen was smarter than that... This is a really quick fix, I'll do it now

Reply:
Actually, no, no time. I'll fix it in the future

Reply:
The problem is not coming from Eigen here but from C++ itself ;)